### ISSUE TITLE
pixelwise-loss-weight map

### ISSUE THREAD

**Issue by shinlin77** [tensorflow/tensorflow#38060]
**System information**
- TensorFlow version (you are using): 1.14.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
U-NET (https://www.nature.com/articles/s41592-018-0261-2) has become a popular way to perform semantic segmentation for cell microscopy.  Instance segmentation requires pixelwise-loss weights.  Currently, model.fit_generator supports only val_sample_weights and class_weight--not pixelwise-loss weights.  

There have been several work-arounds proposed for this issue, and I have implemented almost everyone.  Most generated errors except 

https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/keras-users/ue1S8uAPDKU/PzL7Vb0bBQAJ

It kind of works, but the Jupyter notework keeps issuing warnings like

E tensorflow/core/grappler/optimizers/dependency_optimizer.cc:697] Iteration = 1, topological sort failed with message: The graph couldn't be sorted in topological order.

When I run it, the results change drastically from run to run.

**Will this change the current api? How?**
It will add more versatility to the api by incorporating an important aspect of computer vision.  

**Who will benefit with this feature?**
Those using keras to implement computer vision programs.

**Any Other info.**
Thank you kindly for your consideration.

**Comment by pavithrasv** [comment]
@shinlin77 Will you be able to try this in the latest version of TF? We have made a number of changes to sample weights behavior and we have very few restrictions on sample weights today.
In TF2.2 you are not required to specify `sample_weight_mode` in compile API.

Closing this issue as the use case is expected to work in TF2.0. Please feel free to ping or reopen if you are facing any issues.

**Comment by shinlin77** [comment]
OK, lemme give it a shot. I’ll let you know how it goes. Shin Sent from Mail for Windows 10 From: Pavithra VijaySent: Thursday, April 2, 2020 10:57 AMTo: tensorflow/tensorflowCc: shinlin77; MentionSubject: Re: [tensorflow/tensorflow] pixelwise-loss-weight map (#38060) @shinlin77 Will you be able to try this in the latest version of TF? We have made a number of changes to sample weights behavior and we have very few restrictions on sample weights today.In TF2.2 you are not required to specify sample_weight_mode in compile API.Closing this issue as the use case is expected to work in TF2.0. Please feel free to ping or reopen if you are facing any issues.—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by davideboschetto** [comment]
Sample weights, even if you are using custom generators (as far as I know), need to be 1D or 2D.
For pixelwise weighting, the ideal would be having something like @shinlin77 is saying (for one-hot encoded and NCHW: [batch_size, n_classes, height, width]). 

This at the moment is not possible (sample weights can be either one weight per sample in batch or one weight per class of sample in batch).
